spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -223,9 +223,7 @@ else if (options.isCountingQuery() && this.tree.isDelete()) {
 		ProjectionInformation projectionInformation =
 				this.projectionFactory.getProjectionInformation(this.queryMethod.getReturnedObjectType());
 
-		if (projectionInformation != null &&
```

Comment:
Hmm, how do you know it cannot be null here? Did Sonar make that claim?

Reply:
This one I looked at -- projection factory always gets or creates a projection. Now, it beats me how _Sonar_ knew that, because it's buried a few layers deep in code.

https://github.com/spring-projects/spring-data-commons/blob/master/src/main/java/org/springframework/data/projection/ProxyProjectionFactory.java#L145

Reply:
Yep ^